drivers/digital-io/max22200/max22200.c

```diff
@@ -0,0 +1,958 @@
+ * @param val - Value to be updated in the desired register.
+ * 		Should be bit shifted by using no_os_field_prep(mask, val)
+ * @param one_byte - Value will be updated only on the LSB if true,
+ * 		     on the whole register if false.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_reg_update(struct max22200_desc *desc, uint32_t reg, uint32_t mask,
+			uint32_t val, bool one_byte)
+{
+	int ret;
+	uint32_t reg_val = 0;
+
+	ret = max22200_reg_read(desc, reg, &reg_val, one_byte);
+	if (ret)
+		return ret;
+
+	reg_val &= ~mask;
+	reg_val |= mask & val;
+	return max22200_reg_write(desc, reg, reg_val, one_byte);
+}
+
+/**
+ * @brief Set external trigger's state of the MAX22200
+ * @param desc - MAX22200 device descriptor.
+ * @param trig_state - State to change to.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_set_trig_state(struct max22200_desc *desc,
+			    enum max22200_trig_state trig_state)
+{
+	if (!desc->trig_desc)
+		return -EINVAL;
+
+	switch (trig_state) {
+	case MAX22200_TRIG_OFF:
+		return no_os_gpio_set_value(desc->trig_desc, NO_OS_GPIO_LOW);
+	case MAX22200_TRIG_ON:
+		return no_os_gpio_set_value(desc->trig_desc, NO_OS_GPIO_HIGH);
+	default:
+		return -EINVAL;
+	}
+}
+
+/**
+ * @brief Set channel state for specific channel
+ * @param desc - MAX22200 device descriptor.
+ * @param ch - Channel number.
+ * @param ch_state - Channel's state to be changed into.
+ * @return 0 in case of succces, negative error code otherwise.
+*/
+int max22200_set_ch_state(struct max22200_desc *desc, uint32_t ch,
+			  enum max22200_ch_state ch_state)
+{
+	if (ch > MAX22200_CHANNELS - 1)
+		return -EINVAL;
+
+	return max22200_reg_update(desc, MAX22200_STATUS_REG,
+				   MAX22200_ONCH_MASK(ch),
+				   no_os_field_prep(MAX22200_ONCH_MASK(ch),
+						   ch_state), false);
+}
+
+/**
+ * @brief Set fault mask bits in the status register.
+ * @param desc - MAX22200 device descriptor.
+ * @param fault_mask - Desired fault mask bit to set.
+ * @param enabled - enabled/disabled state of the mask.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_fault_mask_set(struct max22200_desc *desc,
+			    enum max22200_fault_mask fault_mask, bool enabled)
+{
+	if (fault_mask < 17 || fault_mask > 23)
+		return -EINVAL;
+
+	if (fault_mask == MAX22200_M_HHF && desc->chan_drive == MAX22200_VOLTAGE_DRIVE)
+		return -EINVAL;
+
+	return max22200_reg_update(desc, MAX22200_STATUS_REG,
+				   MAX22200_STATUS_FAULT_MASK(fault_mask),
+				   no_os_field_prep(MAX22200_STATUS_FAULT_MASK(fault_mask),
+						   enabled), false);
+}
+
+/**
+ * @brief Set device frequency value.
+ * @param desc - MAX22200 device descriptor.
+ * @param device_freq - Device frequency value.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_set_dev_freq(struct max22200_desc *desc,
+			  enum max22200_device_freq device_freq)
+{
+	if (device_freq > 1)
```

Comment:
since the `device_freq` is of type `enum max22200_device_freq` I don't think this check is useful.

Reply:
removing it.